### ISSUE TITLE
Confirm that we're checking range health during rolling restart

### ISSUE THREAD

**Issue by keith-mcclellan** [cockroachdb/cockroach-operator#364]
Rolling restarts (as done during upgrades and other operations) should confirm that the cluster doesn't have any under-replicated ranges before starting and between restarting each pod to guarantee that the database remains available during these operations.  Currently, we're checking pod and db readiness but not confirming that the cluster is completely healthy.

**Comment by keith-mcclellan** [comment]
Here's an example of injecting this into our statefulset config:

```
       startupProbe:
          exec:
            command:
            - /bin/sh
            - -c
            - |
              for i in {1..30};
              do
                UR=$(/cockroach/cockroach sql \
                   --certs-dir=/cockroach/cockroach-certs/ \
                   -e "SELECT SUM((metrics->>'ranges.underreplicated')::DECIMAL)::INT8 AS ranges_underreplicated FROM crdb_internal.kv_store_status S INNER JOIN crdb_internal.gossip_liveness L ON S.node_id = L.node_id WHERE L.decommissioning <> true;" \
                   --format raw \
                   --host=cockroachdb-public | awk '{if(NR>3)print}' | awk '{if(NR==1)print}'
                     );
                echo "Under-replicated ranges: $UR" >> /usr/share/message;
                if [ -z "$UR" ];
                then
                  echo "No under-replicated ranges reported.  Sleeping for 10 seconds - iteration $i" >> /usr/share/message;
                  sleep 10;
                  continue;
                fi
                if [ $UR -gt 0 ];
                then
                  echo "Sleeping for 10 seconds - iteration $i" >> /usr/share/message;
                  sleep 10;
                else
                  echo "breaking out of loop" >> /usr/share/message;
                  break;
                fi
              done
              exit 0;
          failureThreshold: 1
          periodSeconds: 10
          successThreshold: 1
          timeoutSeconds: 1
```

**Comment by keith-mcclellan** [comment]
shout out to @jhatcher9999 for the initial work on this btw

**Comment by jhatcher9999** [comment]
I think looking at the range replication reports might be a more elegant way to look at this:
```
SELECT crdb_internal.zones.zone_id, target, range_name, database_name, table_name, index_name, under_replicated_ranges
FROM crdb_internal.zones
INNER JOIN system.replication_stats ON crdb_internal.zones.zone_id = system.replication_stats.zone_id
WHERE under_replicated_ranges > 0;
```
or, more simply
```
SELECT SUM(under_replicated_ranges) FROM system.replication_stats;
```

**Comment by alinadonisa** [comment]
> I think looking at the range replication reports might be a more elegant way to look at this:
> 
> ```
> SELECT crdb_internal.zones.zone_id, target, range_name, database_name, table_name, index_name, under_replicated_ranges
> FROM crdb_internal.zones
> INNER JOIN system.replication_stats ON crdb_internal.zones.zone_id = system.replication_stats.zone_id
> WHERE under_replicated_ranges > 0;
> ```
> 
> or, more simply
> 
> ```
> SELECT SUM(under_replicated_ranges) FROM system.replication_stats;
> ```

@keith-mcclellan, @jhatcher9999  so if the result of the SUM(under_replicated_ranges) it is 0 we can move to the next pod?

**Comment by udnay** [comment]
Closed by #527
